### DISCUSSION TITLE
Pass hidden input value that was set by JS

### DISCUSSION THREAD

**Discussion by joostvanhoof** [livewire/livewire#3367] [1. Help]
If I have a component like so:

```php
class Example extends Component
{
    public array $data = [];

    public function submit()
    {
        dd($this->data)
    }
}
```

and a blade component:

```html
<form wire:submit.prevent="submit">

<input type="hidden" id="test1" wire:model="data.test1">
<input type="hidden" id="test2" wire:model="data.test2">

<button type="submit"></button>
</form>
```

and then javascript on that page that fills the values of these hidden input fields:

```javascript
window.addEventListener('load', function() {
  let element1 = document.getElementById('test1')
  let element1.value = 'hi'

  let element2 = document.getElementById('test2')
  let element2.value = 'hi again'
})
```

then the `$this->data` array will be empty on submit. I believe this is because Livewire overwrites these values because of `wire:model`. How can I make sure that the values that are set by the javascript are submitted?

**Comment by joshhanley** [comment]
@joostvanhoof Livewire doesn't support updating the value of an element directly, as Livewire has no way to detect that change. If anything, I would not use a hidden input, and instead just update Livewire's properties directly.

```js
window.addEventListener('load', function() {
  // let element1 = document.getElementById('test1')
  // let element1.value = 'hi'
  @this.data.test1 = 'hi'

  // let element2 = document.getElementById('test2')
  // let element2.value = 'hi again'
  @this.data.test2 = 'hi again'
})
```

Hope this helps!

**Reply by joostvanhoof** [reply]
Thanks @joshhanley, that's a good solution. I guess it only works if the javascript is on the page (not in a bundle) because else `@this` is not aware of the Livewire properties?

In the meantime I found a post on the old Livewire forum that suggested firing an input event which also works, and can be used in a bundle as well:

```javascript
window.addEventListener('load', function() {
  let element1 = document.getElementById('test1')
  let element1.value = 'hi'
  element.dispatchEvent(new Event('input'))

  let element2 = document.getElementById('test2')
  let element2.value = 'hi again'
  element.dispatchEvent(new Event('input'))
})
```

I actually don't know if I need to fire the event after each field value update or just once at the end but it works like this :-)

**Reply by joshhanley** [reply]
@joostvanhoof yeah, you'd have to find the element and the look for the closest Livewire component (instead of `@this` if you wanted to extract the JS.

I can share code on that if your interested.

But firing off an input event works just as well 🙂

**Reply by joostvanhoof** [reply]
Got it, thanks for the help. For others if they read this thread, I think you find the closest Livewire component as such, correct?

```javascript
let livewireComponent = Livewire.find(
  e.target.closest('[wire\\:id]').getAttribute('wire:id')
)
```

How would you then proceed to use parameters of that Livewire component?

**Reply by joshhanley** [reply]
@joostvanhoof yep, although that is if you use it inside an event. If you have it in a page load event listener, as above, you would need to manually find the element and then call closest on that.

One you have the component, you use it the same as you would `@this`. So like `livewireComponent.set('data.test1', 'Hi')`
